File: Interfaz/ProyectoFinal/src/app/Components/TipoDeUsuario/class/tipoDeUsuario.ts
import { Modulo } from "../../Modulo/class/modulo";

export class TipoDeUsuario{
    private idTipoDeUsuario!: number;
 
    private nombre!: string;
    
    private activo!: boolean;
   
    private modulos: Modulo[] = [];

    constructor(){}

    public get IdTipoDeUsuario(): number {
        return this.idTipoDeUsuario;
    }
    public set IdTipoDeUsuario(value: number) {
        this.idTipoDeUsuario = value;
    }
    public get Nombre(): string {
        return this.nombre;
    }
    public set Nombre(value: string) {
        this.nombre = value;
    }
    public get Activo(): boolean {
        return this.activo;
    }
    public set Activo(value: boolean) {
        this.activo = value;
    }
    public get Modulos(): Modulo[] {
        return this.modulos;
    }
    public set Modulos(value: Modulo[]) {
        this.modulos = value;
    }
}

File: ProyectoFinal/ProyectoFinal/2_Dominio/Entidades/TipoDeUsuario.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ProyectoFinal._2_Dominio.Entidades
{
    public class TipoDeUsuario
    {
        public int IdTipoDeUsuario { get; set; }
        public string Nombre { get; set; }
        public bool Activo { get; set; }
        public List<Modulo> Modulos { get; set; }
    }
}